tests/test_cases/test_cocotb/test_scheduler.py

```diff
@@ -206,3 +206,59 @@ async def killer():
     dut.clk <= 1
     await Timer(1)
     assert not victim_resumed
+
+@cocotb.test()
+async def test_nulltrigger_reschedule(dut):
+    """
+    Test that NullTrigger doesn't immediately reschedule the waiting coroutine.
+
+    The NullTrigger will be added to the end of the list of pending triggers.
+    """
+
+    last_fork = None
+
+    @cocotb.coroutine
+    async def reschedule(n):
+        nonlocal last_fork
+        for i in range(4):
+            cocotb.log.info("Fork {}, iteration {}, last fork was {}".format(n, i, last_fork))
+            assert last_fork != n
+            last_fork = n
+            await NullTrigger()
+
+    # Test should start in event loop
+    await Combine(*(reschedule(i) for i in range(4)))
+
+    await Timer(1)
+
+    # Event loop when simulator returns
+    await Combine(*(reschedule(i) for i in range(4)))
+
+@cocotb.test()
+async def test_event_set_schedule(dut):
+    """
+    Test that Event.set() doesn't cause an immediate reschedule.
+
+    The coroutine waiting with Event.wait() will be woken after
+    the current coroutine awaits a trigger.
+    """
+
+    e = Event()
+    event_set = False
```

Comment:
`Event` has a `.fired` attribute, use that instead.

Reply:
It should probably also have a `is_set()` method to mirror asyncio.

Reply:
I actually want to check if the other coro/task wakes up. I'll fix the naming.